src/main/java/com/pinterest/secor/uploader/Uploader.java

```diff
@@ -233,6 +233,8 @@ protected void checkTopicPartition(TopicPartition topicPartition) throws Excepti
                 LOG.debug("previous committed offset count {} is lower than committed offset {} is lower than or equal to last seen offset {}. " +
                                 "Trimming files in topic {} partition {}",
                         oldOffsetCount, newOffsetCount, lastSeenOffset, topicPartition.getTopic(), topicPartition.getPartition());
+                if (System.getProperty("exitOnTrim") != null)
+                    System.exit(43);
```

Comment:
Could you add a comment what's the goal of killing the process?

Reply:
+1